### ISSUE TITLE
Add 'always a wire' clock crossings

### ISSUE THREAD

**Issue by JulianKemmerer** [JulianKemmerer/PipelineC#68]
Current clock crossings take into account clock ratios to buffer and/or require fifos if the clock domains are not the same.

Create a way of declaring a wire-always clock domain crossing (always a wire, not just when same clock domain - for all domain crossings).

**Comment by JulianKemmerer** [comment]
This should almost certainly be called an 'async' wire or something
People cant assume directly putting a wire across the clock domain will just work to transfer data.

**Comment by JulianKemmerer** [comment]
Current clock crossing info thinking about 
https://github.com/JulianKemmerer/PipelineC/wiki/Main-Function-Clock-Crossings

**Comment by JulianKemmerer** [comment]
The tool default makes clock crossing logic to match the ratio between the user clocks. This isssue is an override of default behavior - so seems sorta reasonable to **make a pragma** - as opposed to newly named kind of clock crossing. Maybe reconsider not being pragma is clock crossings are every declared explictly as what type (wire,fifo,ratio'd de/serializers).

**Comment by JulianKemmerer** [comment]
~ async pragma for 
```
Non-Volatile Clock Crossings
These shouldn't really be thought of as clock crossings since non volatile here means that the clocks are synchronous (specifically same freq for now, R=1 only). Such 'crossings' can really be thought of as arbitrary direct connections between main functions. These can be used for exposing modules with globally visible ports/wires.
```

**Comment by JulianKemmerer** [comment]
```c
type_t the_wire;
#include "clock_crossing/the_wire.h"
#pragma ASYNC_WIRE the_wire
```

If this becomes weird it can be addressed in #72

**Comment by JulianKemmerer** [comment]
https://github.com/JulianKemmerer/PipelineC/commit/6f79b91ac1f54376c82f3fbdd71463e70e3a2b5f
